vault-upgrade-hook/src/main/java/biz/netcentric/vlt/upgrade/util/ResourceSerializer.java

```diff
@@ -0,0 +1,16 @@
+/*
+ * (C) Copyright 2016 Netcentric AG.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package biz.netcentric.vlt.upgrade.util;
+
+import org.apache.sling.api.resource.Resource;
+
+public interface ResourceSerializer {
```

Comment:
Do we really need this interface? I don't see it used anywhere.

Reply:
I would prefer to program to an interface, for future extensions. There is no declared variable of this type only because resource serializer is needed in the first row of constructor call (super() call), and I can not do this assignment. 

So currently it's mostly needed for the feature if, for example, we will need to replace json with xml serialization for some reason. Then we would have an interface to implement.

Does this sound reasonable?
Otherwise, it's easy to get rid of it.

Reply:
Too many "if"s there in my opinion. In case we really need it someday we can still add an interface, until then please remove it.